lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -3340,8 +3359,11 @@ class ObjCMethodConventions : public Conventions {
     if (isImplicitPlusOneCFResult())
       return ResultConvention::Owned;
 
-    if (tl.getLoweredType().isForeignReferenceType())
+    if (tl.getLoweredType().isForeignReferenceType()) {
+      if (auto resultConventionOpt = getCxxRefConventionWithAttrs(tl, Method))
+        return *resultConventionOpt;
       return ResultConvention::Unowned;
+    }
```

Comment:
nit: if you care for minimizing the diff + making this a one-liner:

```suggestion
    if (tl.getLoweredType().isForeignReferenceType())
      return getCxxRefConventionWithAttrs(tl, Method).value_or(ResultConvention::Unowned);
```